### ISSUE TITLE
[HOLD for payment 2023-05-31] [$4000] App does not hold scroll position and scrolls to top in LHN on tapping back button from chat page

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#14125]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Open the app
2. Scroll to bottom and open any report
3. Swipe or click on back button to get back to archive page

## Expected Result: 
Scroll position should be same as it was before entering into the report

## Actual Result: 
App scrolls to top after getting back to report archive page from a report

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native

**Version Number:** 1.2.50-13
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/211229560-9fbd3180-f4d4-4504-8616-047d8aef15bb.mp4

https://user-images.githubusercontent.com/43996225/211229581-005cbf05-0c74-4ac6-8f1c-184443807789.mp4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @Dhanashree-Sawant
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1673087531900939

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~019f5140c7ae3d73d0</li>
        <li>Upwork Job ID: 1612519861034061824</li>
        <li>Last Price Increase: 2023-01-24</li>
    </ul>
</details>

**Comment by tjferriss** [comment]
I was able to reproduce this using BrowserStack.

**Comment by tjferriss** [comment]
>If it's an upstream issue, let's fix it upstream.

@rushatgabhane can you help me understand what needs to happen here? Should we still be looking for proposals for this issue?

**Comment by tjferriss** [comment]
@rushatgabhane @iwiznia can you help me out here with what needs to happen? Is the proposal to close this issue in favor of https://github.com/software-mansion/react-freeze/issues/31?

**Comment by iwiznia** [comment]
I think no, since that issue might not get worked on. But yes, we need to be solving that issue and sending a PR upstream.

**Comment by tjferriss** [comment]
Ok so should we put this issue on hold until we have the upstream fix?

**Comment by s77rt** [comment]
@tjferriss The upstream fix is supposed to originate from this issue. It won't make sense to put this on hold.

**Comment by situchan** [comment]
I think we can put this on hold for navigation project since freeze will be deprecated.
ref: https://github.com/Expensify/App/issues/13947#issuecomment-1399036991

**Comment by MelvinBot** [comment]
@iwiznia @tjferriss @rushatgabhane this issue is now 4 weeks old and preventing us from maintaining WAQ, can you:
- Decide whether any proposals currently meet our guidelines and can be approved as-is today
- If no proposals meet that standard, please take this issue internal and treat it as one of your highest priorities
- If you have any questions, don't hesitate to start a discussion in #expensify-open-source

Thanks!

**Comment by MelvinBot** [comment]
Current assignee @rushatgabhane is eligible for the Internal assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
@iwiznia @tjferriss @rushatgabhane this issue is now 4 weeks old and preventing us from maintaining WAQ. This should now be your highest priority. Please post below what your plan is to get a PR in review ASAP. Thanks!

**Comment by MelvinBot** [comment]
@iwiznia @tjferriss @rushatgabhane this issue is now 4 weeks old and preventing us from maintaining WAQ. This should now be your highest priority. Please post below what your plan is to get a PR in review ASAP. Thanks!

**Comment by tjferriss** [comment]
My understanding is that this issue is still on hold while we wait for https://github.com/Expensify/App/issues/13947 which is also on hold for the navigation reboot project. So I'm going to move down the priority to weekly.

**Comment by iwiznia** [comment]
Why was this marked as overdue if it is on hold?! 😕

**Comment by tjferriss** [comment]
Still on hold

**Comment by tjferriss** [comment]
Still on hold

**Comment by tjferriss** [comment]
Still on hold

**Comment by iwiznia** [comment]
Still on hold

**Comment by iwiznia** [comment]
Still on hold

**Comment by tjferriss** [comment]
Easy Melvin. This is still on hold.

**Comment by tjferriss** [comment]
Easy Melvin. This is still on hold.

**Comment by mountiny** [comment]
I can confirm this should be fixed with the refactor on native devices

**Comment by iwiznia** [comment]
Should we close this then?

**Comment by mountiny** [comment]
Shall we pay @dhanashree-sawant for reporting the bug?

Sharing a videos of the working behaviour on the navigation refactor branch:

iOS:

https://github.com/Expensify/App/assets/36083550/3f5d9d68-913a-4cbd-9bc4-09fbc03ee6d9

Safari mWeb:

https://github.com/Expensify/App/assets/36083550/6596b800-aba3-4291-984d-9d5a4e9ec8eb

**Comment by mountiny** [comment]
@tjferriss Could we please pay out $250 for reporting this bug to @dhanashree-sawant?

It will be fixed with the navigation refactor as you can see in the videos above

**Comment by tjferriss** [comment]
Yes, the offer has been sent to @dhanashree-sawant for the reporting bonus: https://www.upwork.com/ab/applicants/1662145730731999232/job-details

**Comment by mountiny** [comment]
So close to close this one out 😍

**Comment by tjferriss** [comment]
@dhanashree-sawant has been paid.

**Comment by ahmedGaber93** [comment]
Coming from https://github.com/Expensify/App/issues/84362, we identified an edge case where the `contentOffset` is not saved correctly. This occurs after forcing `scrollToEnd` or `scrollToOffset` with `animated: false` (e.g., `scrollViewRef.current.scrollToEnd({ animated: false });`). In this scenario, `onMomentumScrollEnd` is not triggered. To handle this, we relied on `onScroll`, and confirmed that the original issue is not reproducible.
